mops/include/mops/cuda_first_occurences.hpp

```diff
@@ -0,0 +1,18 @@
+#ifndef FIRST_OCCURENCES_HPP
+#define FIRST_OCCURENCES_HPP
+
+#include "mops/tensor.hpp"
+#include <cstdint>
+
+/*
+ * Computes the indexes at which the sorted input array (receiver_list) change in value, with 0
+ * prepended. For example, if the receiver list is [0, 0, 0, 1, 1, 1, 2, 2], then the output would
+ * be [0, 3, 6]. nelements_input refers to the size of the input receiver_list, nelements_output
+ * refers, to the number of output elements. The elements of receiver_list **must** be sorted such
+ * that all references to each index appear contiguously and continuously.
+ */
```

Comment:
```suggestion
 *
 * This function allocates memory which must be freed with `cudaFree` when no longer required.
 */
```

This is how I understand it, is it right?

Reply:
yes that's correct.